Rework Introduction (#144)
decentralized-identity/bbs-signature: draft-bbs-signatures.md

A digital signature scheme is a fundamental cryptographic primitive that is used to provide data integrity and verifiable authenticity in various protocols. The core premise of digital signature technology is built upon asymmetric cryptography where-by the possessor of a private key is able to sign a message, where anyone in possession of the corresponding public key matching that of the private key is able to verify the signature.

Beyond the core properties of a digital signature scheme, BBS signatures provide multiple additional unique properties, three key ones are:

> **Selective Disclosure** - The scheme allows a signer to sign multiple payloads and produce a single output signature. An intermediary (prover) then possessing the payloads and the signature can generate a proof whereby they can choose which payloads to reveal.

Author: Yeap sure good catch

> **Selective Disclosure** - The scheme allows a signer to sign multiple payloads and produce a single output signature. An intermediary (prover) then possessing the payloads and the signature can generate a proof whereby they can choose which payloads to reveal.

Author: ```suggestion
**Selective Disclosure** - The scheme allows a signer to sign multiple messages and produce a single output signature. An intermediary (prover) then possessing the messages and the signature can generate a proof whereby they can choose which messages to disclose.
```

> **Selective Disclosure** - The scheme allows a signer to sign multiple messages and produce a single output signature. An intermediary (prover) then possessing the messages and the signature can generate a proof whereby they can choose which messages to disclose.

Author: Can we tweak the language from revealed -> disclosed?

> **Selective Disclosure** - The scheme allows a signer to sign multiple messages and produce a single output signature. An intermediary (prover) then possessing the messages and the signature can generate a proof whereby they can choose which messages to disclose.

Reviewer: Yeap! Updated the suggestion

> **Selective Disclosure** - The scheme allows a signer (issuer) to sign multiple messages and produce a single -constant size- output signature. An intermediary (prover) then possessing the messages and the signature can generate a proof whereby they can choose which messages to disclose, while leaking no-information about the un-disclosed messages. The proof itself guarantees the integrity and authenticity of the disclosed messages (e.g., that they were originally signed by the issuer).

Author: ```suggestion
**Selective Disclosure** - The scheme allows a signer (issuer) to sign multiple messages and produce a single -constant size- output signature. An intermediary (prover) then possessing the messages and the signature can generate a proof whereby they can choose which messages to disclose, while leaking no-information about the un-disclosed messages. The proof itself guarantees the integrity and authenticity of the disclosed messages (e.g. that they were originally signed by the issuer).
```

**Unlinkable Proof Presentations** - The proofs generated by the scheme are known as a signature proof of knowledge, meaning a verifying party in receipt of a proof is unable to determine which signature was used to generate the proof, removing a common source of correlation. In general each proof generated is indistinguishable from random even for two proofs generated from the same signature.

> **Proof of Posession** - The proofs generated by the scheme prove to a verifier that the party who generated the proof (prover) was in posession of a signature without revealing it. Because the scheme also supports binding a presentation header to the generated proof it makes the scheme suitable as a proof of posession protocol. Note the presentation header can include abitrary information such as a cryptographic nonce and or an audience/domain identifier to ensure the generated proof can only be used appropriately, including providing a way for a verifier to detect a replay attack.

Reviewer 2: *possession :)

Author: ```suggestion
**Proof of Possession** - The proofs generated by the scheme prove to a verifier that the party who generated the proof (prover) was in possession of a signature without revealing it. Because the scheme also supports binding a presentation header to the generated proof it makes the scheme suitable as a proof of possession protocol. Note the presentation header can include arbitrary information such as a cryptographic nonce and or an audience/domain identifier to ensure the generated proof can only be used appropriately, including providing a way for a verifier to detect a replay attack.
```

> **Proof of Possession** - The proofs generated by the scheme prove to a verifier that the party who generated the proof (prover) was in possession of a signature without revealing it. Because the scheme also supports binding a presentation header to the generated proof it makes the scheme suitable as a proof of possession protocol. Note the presentation header can include arbitrary information such as a cryptographic nonce and or an audience/domain identifier to ensure the generated proof can only be used appropriately, including providing a way for a verifier to detect a replay attack.

Reviewer: ```suggestion
**Proof of Possession** - The proofs generated by the scheme prove to a verifier that the party who generated the proof (prover) was in possession of a signature without revealing it. The scheme also supports binding a presentation header to the generated proof. The presentation header can include arbitrary information such as a cryptographic nonce, an audience/domain identifier to ensure the generated proof can only be used appropriately, including providing a way for a verifier to detect a replay attack, or any other information a protocol may require to be cryptographically tied to a proof.
```
Since the BBS+ proof is by definition a proof of possession protocol even without the presentation header the above could be a litle less confusing??

Author: Agreed
